dali/pipeline/operators/crop/slice.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/crop/slice.h"
+
+namespace dali {
+
+DALI_SCHEMA(Slice)
+    .DocStr(R"code(Crop as slice of a defined `size` from an `input` tensor, staring
+    at the location specified by `begin`. Inputs must be supplied as 3 Tensors in a
+    specific order: `Images` containing image data in NHWC format, `Begin` containing
+    the starting pixel coordinates for the `crop` in `(x,y)` format, and 'Size' containing
+    the pixel dimensions of the `crop` in `(w,h)` format. The resulting tensor output of
+    Slice operation is a cropped version of the input tensor `Images`.)code")
+    .NumInput(3)
+    .NumOutput(1)
+    .AllowMultipleInputSets()
+    .EnforceInputLayout(DALI_NHWC)
+    .AddParent("Crop");
+
+template <>
+void Slice<CPUBackend>::DataDependentSetup(SampleWorkspace *ws) {
+  // Assumes xywh. ltrb not supported atm
+  const auto &input = ws->Input<CPUBackend>(0);
+  const auto &begin = ws->Input<CPUBackend>(1);
+
+  const int H = input.shape()[0];
+  const int W = input.shape()[1];
+
+  crop_y_norm_[ws->data_idx()] = static_cast<float>(begin.template data<float>()[1] / H);

```

Comment:
crop_y_norm_[ws->data_idx()] = static_cast<float>(begin.template data<float>()[ws->data_idx()*2 + 1] / H);
crop_x_norm_[ws->data_idx()] = static_cast<float>(begin.template data<float>()[ws->data_idx()*2] / W);
?

Reply:
As spoken, what you propose would be the best for tensor lists, however CPUBackend uses tensors